### PR TITLE
Reorganize helpers functions

### PR DESCRIPTION
**Pull request by aliculPix4D**
- pr_info, recursive_path and auto_merge are now part of `helpers_functions.rb`
- enables us to write tests for these functions and reduces the amount of files

### CODE REVIEW

**Comment by aledeganopix4d** [comment]
Nice refactoring!

**Comment by marco-m-pix4d** [comment]
From the high level, I have mixed feelings here.

I understand and applaud the intent and the fact of being able to write tests.

But, the previous files had a specific name, while now we have a generic name `helpers_functions` that is the same as having a name such as `foo` or `bag_of_stuff`.

What I wonder is if there is a way to reorganize the code, that makes it testable and at the same time has file names that tell a story about what is inside.

Where the names of the deleted files wrong ?

Also, "reducing the amount of files" could be better or worse, it depends on the situation.

**Comment by odormond** [comment]
> But, the previous files had a specific name, while now we have a generic name `helpers_functions` that is the same as having a name such as `foo` or `bag_of_stuff`.

A slightly better name would be `github_helpers` because all the functions in there are tight to GitHub.

**Comment by aliculPix4D** [comment]
> From the high level, I have mixed feelings here.
> 
> I understand and applaud the intent and the fact of being able to write tests.
> 
> But, the previous files had a specific name, while now we have a generic name `helpers_functions` that is the same as having a name such as `foo` or `bag_of_stuff`.
> 
> What I wonder is if there is a way to reorganize the code, that makes it testable and at the same time has file names that tell a story about what is inside.
> 
> Where the names of the deleted files wrong ?
> 
> Also, "reducing the amount of files" could be better or worse, it depends on the situation.

My goal was to add tests. In previous cases, where I was just adding the functions to `dependabot-main.rb`script, that was not possible. Therefore, I went with approach of combining all the files that we introduced and had only a single function, to a single file. The other approach would be to leave the previous files intact (if no changes are required) and create a new file and spec for the extra function. Would something like that work for you? For me it is the same, as long as we can write tests for each of the function.

File: docker/dependabot-main.rb [hunk 1]
```diff
@@ -139,9 +112,16 @@ def auto_merge(pull_req)
 
     pull_request = pr.create
 
-    auto_merge(pull_request)
+    puts pull_request.html_url
+
+    auto_merge(pull_request.number,
```
Comment (muzaffar-omer) [inline docker/dependabot-main.rb]: Are we using only `auto_merge` function from the `helpers_functions` file? ... or other functions are used in different files? ... if we are not using other functions maybe we could delete them from the `helpers_functions` file

Comment (aliculPix4D) [reply docker/dependabot-main.rb]: We are using all of them:
L27 `pr_info(updated_deps.first)`
L87 `input_files_path = recursive_path(...)`

Comment (muzaffar-omer) [reply docker/dependabot-main.rb]: Ahaa, sorry, I did not observe them there, thanks :-)

**Comment by marco-m-pix4d** [comment]
@aliculPix4D It seems that we are talking past each other :-)
Feel free to merge.

**Comment by aliculPix4D** [comment]
> @aliculPix4D It seems that we are talking past each other :-)
> Feel free to merge.

Maybe second attempt is better now:
- no change to `pr_info.rb`, `pr_info_spec.rb` and `path_level.rb` 
- added tests for `path_level.rb` 
- moved def auto_merge() function to separate file `auto_merge.rb` and added tests for it `auto_merge_spec.rb`
- minimal modification were needed in `docker/dependabot-main.rb`

**Comment by Pix4D-Janus** [comment]
Your pipeline is set.
Check it here:https://builder.ci.pix4d.com/teams/developers/pipelines/dependabot-helpers_functions

**Pull request merged**